src/pages/accounts/SignUpPage.tsx

```diff
@@ -47,44 +48,94 @@ const TextLink = styled(TextButton)`
 	word-wrap: break-word;
 `;
 
+interface InputData {
+	value: string;
+	error: boolean;
+}
+
+function inputReducer(state: InputData, action: Partial<InputData>) {
+	return { ...state, ...action };
+}
+
 function validEmail(email: string) {
 	console.log('Checking email');
 	return !!email.match(/.+@.+\..{2,}/);
 }
 
-function compareEmail(email1: string, email2: string) {
-	return !!email1.match(email2);
-}
-
 function validPass(password: string) {
 	if (!password.match(/.{8,}/)) return false; // 8 Characters
 	if (!password.match(/[A-Z]/)) return false; // Upper-Case
 	if (!password.match(/[a-z]/)) return false; // Lower-Case
 	return true; // Passed all checks
 }
```

Comment:
The auth flow looks really cool! A small issue that I had happen when I played around with it is that the first password text input box stays in 'error mode' after a faulty password is typed in once and doesn't revert back to being in its regular state even when the input is changed. Don't think this applies to this ticket, but I thought that I should mention it just in case :)

Reply:
That is definitely part of this ticket! Will fix!